lib/services/pool/lib/pool-composable-join.service.ts

```diff
@@ -381,7 +381,7 @@ export class PoolComposableJoinService {
             if (token.__typename === 'GqlPoolTokenLinear' || token.__typename === 'GqlPoolTokenPhantomStable') {
                 //This token is a nested BPT, not a mainToken
                 //Replace the amount with the chained reference value
-                const index = assets.findIndex((asset) => asset.toLowerCase() === token.address) || -1;
+                const index = assets.findIndex((asset) => asset.toLowerCase() === token.address);
 
                 //if the return amount is 0, we dont pass on the chained reference
                 if (index === -1 || deltas[index] === '0') {
```

Comment:
On line 384 `findIndex` might return `0` as a valid index for the BPT. But this is ofc falsey so the logical operator ends up returning `-1`. And because of that the amountIn for the BPT gets set to `0` instead of the chained reference (see lines 388-391 below)